nengo/conftest.py

```diff
@@ -210,12 +231,17 @@ def pytest_runtest_setup(item):
                 ('logger', 'logs', "logs not requested")]:
             if fixture_name in item.fixturenames:
                 if item.config.getvalue(option):
-                    skip = False
+                    break
                 else:
                     skipreasons.append(message)
-        if skip:
+        else:
             pytest.skip(" and ".join(skipreasons))
 
+    if 'Simulator' in item.fixturenames:
+        for test, reason in _Simulator.unsupported:
```

Comment:
By default, `re` caches 100 patterns, so for most simulators we're good. This value is modifiable, but it's a private variable (`re._MAXCACHE`; doing `re._MAXCACHE = 1000` raises the limit to 1000 patterns). Could be worth setting it to approximately the number of tests we have? In a `try/except` since it's a private variable.